mesh/pncmesh.cpp

```diff
@@ -1119,12 +1137,18 @@ void ParNCMesh::GetFaceNeighbors(ParMesh &pmesh)
             const DenseMatrix* pm = full_list.point_matrices[sf.geom][sf.matrix];
             if (!sloc && Dim == 3)
             {
-               // TODO: does this handle triangle faces correctly?
-
                // ghost slave in 3D needs flipping orientation
                DenseMatrix* pm2 = new DenseMatrix(*pm);
-               std::swap((*pm2)(0,1), (*pm2)(0,3));
-               std::swap((*pm2)(1,1), (*pm2)(1,3));
+               if (sf.geom == Geometry::Type::SQUARE)
+               {
+                  std::swap((*pm2)(0, 1), (*pm2)(0, 3));
+                  std::swap((*pm2)(1, 1), (*pm2)(1, 3));
+               }
+               else if (sf.geom == Geometry::Type::TRIANGLE)
+               {
+                  std::swap((*pm2)(0, 0), (*pm2)(0, 1));
+                  std::swap((*pm2)(1, 0), (*pm2)(1, 1));
```

Comment:
I think this flip is incorrect. Can we add unit tests for these? (Test coverage seems to be missing already for the Hex case)

Reply:
I had been working with the face orientation being 0 as defined on [this line](https://github.com/mfem/mfem/pull/3587/files#diff-c43ff7c21ae1304f127c667baab4061293c38b46496d8603c7e291bc9f6cbac7R1114-R1115). Then going to `geom.cpp`, to swap from orientation 0 to 1, requires swapping the first two columns, assuming the same even-odd patterning as discussed [on the ticket](https://github.com/mfem/mfem/issues/2666#issuecomment-984224259). 

I wasn't sure of how to meaningfully add unit tests past the NCMesh test I put in here, which i acknowledge isn't very satisfactory either. A similar one could be done for hexes, but addressing that wasn't the goal of this PR, as I was just wanting to fix up tets.

Reply:
Probably I misunderstood something in the discussion, but I thought that in [this comment](https://github.com/mfem/mfem/issues/2666#issuecomment-984923708) stated that the orientations in MFEM are defined by even/odd pairs of `Constants<T>::Orient`, i.e.
 https://github.com/mfem/mfem/blob/539f663fe6ad3738b6de9615f6c74dfa44927578/fem/geom.cpp#L909-L913
for triangles. My try can be found in this commit: https://github.com/mfem/mfem/commit/ec7a37b167452b272e794580c8b35082d9bde0cd 

Not arguing that you should also add tests for the hex here, sorry for the misunderstanding. I really just want to make sure to get this flip right with some minimal test (to possibly catch regressions), but I also do not have a really good idea how a proper test could look.

Reply:
I originally had an implementation with the switching, but if you look at the definition of `fi.Elem2Inf` in the lines preceding this section, it is defined as `64 * Elem2No`, i.e. the orientation is always zero after the modulo 64. So I reduced the complexity and just deleted all the switch statements as they couldn't be triggered anyway.

Reply:
I thought it is set this way because the swap in these lines guarantees that we have this orientation (see the code comment below in this file)?

Btw can we for future purposes add an else with an error message here?